code/modules/surgery/bodyparts/bodypart_effects.dm

```diff
@@ -0,0 +1,138 @@
+/// For scaling the effectiveness of certain effects to the total bodypart count
+#define GET_BODYPART_COEFFICIENT(X) round(X.len / 6, 0.1)
+/// Check if it's full body. These are mostly here so we can change just one place when we ever add more limbs (?)
+#define IS_FULL_BODY(X) (X.len == 6)
+
+/// Effects added to a carbon focused on the bodyparts itself, such as adding a photosynthesis component that
+/datum/bodypart_effect
+	/// List of bodyparts contributing to this effect
+	var/list/bodyparts = list()
+	/// Minimum amount of bodyparts required for on_apply to be called. When tipping below, on_remove is called
+	var/minimum_bodyparts = 1
+	/// Are we currently active? We don't NEED to track it, but it's a lot easier and faster if we do
+	var/is_active = FALSE
+	/// Whether or not to hook into COMSIG_LIVING_LIFE and use /proc/on_life()
+	var/process_on_life = FALSE
+
+/datum/bodypart_effect/New(mob/living/carbon/carbon, obj/item/bodypart/bodypart)
+	add_bodypart(carbon, bodypart)
+
+	if(process_on_life)
+		RegisterSignal(carbon, COMSIG_LIVING_LIFE, PROC_REF(on_life))
```

Comment:
Going into this I had expected this datum to be instanced per bodypart, and then hook ``/obj/item/bodypart/proc/on_life(seconds_per_tick, times_fired)`

But now that I see that it's just one datum created per mob I really agree with Jaquerel in that I think this should just use status effects  

In fact this has a shocking similarity to the system genetics uses (`/datum/status_effect/organ_set_bonus`)